### PR TITLE
remove usage of ANY in dtd

### PR DESCRIPTION
**Pull request by ratmice**
This patch restricts mixed-content elements from ANY
to the various leaf nodes which are used.

This is intended to allow for easier parsing in languages with
no recursive datatypes like rust.
Since they no longer have to handle things like
```
<description> text
   <description>more text</description>
<description>
```

which the dtd allowed for, but never used.

tested with:
```
for i in {libsel4/arch_include/*/interfaces/,\
libsel4/include/interfaces/,\
libsel4/sel4_arch_include/*/interfaces}/*.xml;
do
     xmllint --noout --dtdvalid libsel4/tools/sel4_idl.dtd $i;
done
```

### CODE REVIEW

File: libsel4/tools/sel4_idl.dtd [hunk 1]
```diff
@@ -26,23 +26,23 @@
...
 <!ATTLIST error description CDATA #IMPLIED>
 
-<!ELEMENT brief ANY>
+<!ELEMENT brief (#PCDATA|texttt|docref|shortref|autoref|obj)*>
 
-<!ELEMENT description ANY>
+<!ELEMENT description (#PCDATA|texttt|docref|shortref|autoref|obj)*>
 
-<!ELEMENT return ANY>
+<!ELEMENT return (#PCDATA|texttt|docref|shortref|autoref|obj|errorenumdesc)*>
```
Comment (ratmice) [inline libsel4/tools/sel4_idl.dtd:43]: Wasn't sure if `errorenumdesc` belonged in the other ones or not it wasn't used anywhere but `return`.

**Review by lsf37 — APPROVED** [review]
Looks good from my side, and reminds me that the xml lint check should also run on dtd changes..

**Comment by lsf37** [comment]
Could you please a Signed-off-by for the commit?

**Comment by ratmice** [comment]
added signed-off-by, and updated to the lastest.

**Comment by ratmice** [comment]
So I noticed one more thing (I had accidentally added self-recursion in `docref`, updated patch removes that.

File: libsel4/tools/sel4_idl.dtd (hunk 1 above)
Comment (lsf37) [reply libsel4/tools/sel4_idl.dtd:43]: I think it makes sense to restrict it to this one. We can always add it elsewhere if we decide it is necessary.

**Review by lsf37 — APPROVED** [review]

**Comment by ratmice** [comment]
Since i've made a few changes since starting this,
I wanted to confirm that I believe this now does everything I set out to:

I put my rust parser which relies on this formulation of the dtd here: https://github.com/ratmice/sel4_rust_tools
It could use some more testing, and isn't pretty, but should follow the dtd closely.

**Pull request merged**